analyse_purchases.py

```diff
@@ -36,8 +32,8 @@ def analyse_and_print_observations(df, metric, SEGMENT_CATEGORIES, NUMERATOR_COL
                                                                                                   DENOMINATOR_COLUMN)
 
     ## Creatig comparision columns for filtering
-    metric_analysis_df["value_change_wrt_overall"] = (metric_analysis_df[
-                                                          NUMERATOR_COLUMN] * 100) / overall_absolute_change
+    metric_analysis_df["value_change_wrt_overall"] = ((metric_analysis_df[NUMERATOR_COLUMN] - metric_analysis_df[
+        DENOMINATOR_COLUMN]) * 100) / overall_absolute_change
```

Comment:
Suggestion: Inculcate a function over here to deal with edge cases like division by zero.

Reply:
Noted